proxy/proxy.go

```diff
@@ -201,11 +206,47 @@ func forward(r *http.Request) (*http.Response, error) {
 
 func (p *Proxy) proxyResponse(w *ResponseWriter, r *ResponseReader, headers http.Header) error {
 	w.takeHeaders(r)
-	transcoder, found := p.transcoders[r.ContentType()]
+	content_type := r.ContentType()
+	transcoder, found := p.transcoders[content_type]
 	if !found {
 		return w.ReadFrom(r)
 	}
 	w.setChunked()
+
+	body, err := httputil.DumpResponse(r.r, true)
+	body_copy := body
+	if err != nil {
+		log.Fatal(err)
+	}
+	resp_reader := bufio.NewReader(bytes.NewReader(body))
+	resp_reader_copy := bufio.NewReader(bytes.NewReader(body_copy))
+	resp, err := http.ReadResponse(resp_reader, nil)
+	resp_copy, err := http.ReadResponse(resp_reader_copy, nil)
+	body_data, err := io.ReadAll(resp_copy.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	real_content_type := ""
+	if body_data[0] == 0xff && body_data[1] == 0xd8 {
+		real_content_type = "image/jpeg"
+	}
+	if body_data[0] == 0x89 && body_data[1] == 0x50 && body_data[2] == 0x4E &&
+		body_data[3] == 0x47 && body_data[4] == 0x0D && body_data[5] == 0x0A && body_data[6] == 0x1A {
+		real_content_type = "image/gif"
+	}
+	if body_data[0] == 0x47 && body_data[1] == 0x49 && body_data[2] == 0x46 {
+		real_content_type = "image/png"
+	}
+	if real_content_type != "" && real_content_type != content_type {
```

Comment:
What kind of test can you add for this behavior?

Reply:
Hi gaul,
we find that quite a few tyros may incorrectly configure the 'content-type' field in HTTP header.
There are many discussions about this stuff on the Internet.
Just to list a few:
https://stackoverflow.com/questions/69890597/web-server-content-type-not-configuring-correctly-on-apache
https://www.drupal.org/project/drupal/issues/1440534

This drives us to think about making compy more robust and friendly to these web pages with the above modifications :)

Reply:
Are there any tests you can add to `compy_test.go`?